libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
```

Comment:
The equivalent in ToneAlarm is done in the `init()` call - could this be done there?

Reply:
Putting 'play_pattern(SINGLE_BUZZ)' at the end of 'init()' wasn't working (no sound), I think because too much time would elapse before the first invocation of 'Buzzer::update()'.  Moving the '_pattern_start_time = AP_HAL::millis()' assignment into 'update_playing_pattern()' gets it working, and is probably better in terms of proper timing.